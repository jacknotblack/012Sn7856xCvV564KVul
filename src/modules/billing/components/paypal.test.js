import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Paypal from "./paypal";

describe("A suite", () => {
  let wrapper;
  const props = {
    onPaypalScriptLoaded: jest.fn(),
    onPaypalScriptLoadFail: jest.fn(),
    paypalStatus: "ready to paypal",
    paypalPayment: "some payment detail",
    createPayment: jest.fn(),
    onPaypalAuthorized: jest.fn(),
    cancelPayment: jest.fn()
  };

  beforeEach(() => {
    wrapper = shallow(<Paypal {...props} />);
    window.paypal = {
      Button: {
        render: () => {
          wrapper.instance().handleScriptError("some error");
        }
      }
    };
  });

  describe("render", () => {
    it("renders without crashing", () => {
      expect(wrapper.length).toEqual(1);
    });

    it("should match snapshot", () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it("should render paypal status", () => {
      expect(wrapper.find(".paypal-status").text()).toEqual(
        `STATUS: ${props.paypalStatus} `
      );
    });

    it("should render paypal payment details", () => {
      expect(wrapper.find(".payment-detail").text()).toEqual(
        JSON.stringify(props.paypalPayment)
      );
    });
  });

  describe("scripts", () => {
    it("handles paypal script loaded", () => {
      wrapper.instance().handleScriptLoad();
      expect(props.onPaypalScriptLoaded.mock.calls.length).toEqual(1);
    });

    it("handles paypal script load Error", () => {
      wrapper.instance().handleScriptError();
      expect(props.onPaypalScriptLoadFail.mock.calls.length).toEqual(2);
    });
  });
});
