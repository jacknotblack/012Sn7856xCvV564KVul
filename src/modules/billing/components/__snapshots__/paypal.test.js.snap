// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A suite render should match snapshot 1`] = `
<div
  className="paypal"
>
  <Script
    attributes={Object {}}
    onCreate={[Function]}
    onError={[Function]}
    onLoad={[Function]}
    url="https://www.paypalobjects.com/api/checkout.js"
  />
  <div
    id="paypal-button"
  />
  <h1
    className="paypal-status"
  >
    STATUS: 
    ready to paypal
     
  </h1>
  <span
    className="payment-detail"
  >
    "some payment detail"
  </span>
</div>
`;
